### PR TITLE
Update neovim path to be machine-overridable for remote use

### PR DESCRIPTION
**Pull request by xconverge**
This is really minor but at least is a slight improvement with no downsides

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -354,7 +354,7 @@
       }
     ],
     "configuration": {
-      "title": "Vim Configuration",
```
Comment (xconverge) [inline package.json]: I noticed in my settings extension list, we were the odd-man out here with this verbosity

**Comment by J-Fields** [comment]
We've got `"extensionKind": "ui"` in package.json. I thought this meant that the extension runs on the user's machine, not remotely, and so this wouldn't benefit us? Only vaguely familiar with this stuff, though, so I could be wrong.

**Comment by xconverge** [comment]
Yea the documentation for machine-overridable is not great, I am fine if we skip this for now, I just figured in the offchance it helps someone I couldn't see any harm.

**Comment by jpoon** [comment]
LGTM. Ship it.

**Comment by xconverge** [comment]
Sure, not fully sure what it does, but pretty sure it doesn't hurt. So if it helps someone then 🤷‍♂

**Pull request merged**